compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -29,24 +29,43 @@ namespace {
 class PolynomialApproximationPass final
     : public impl::PolynomialApproximationPassBase<
           PolynomialApproximationPass> {
+public:
+  using Base::Base;
+
   void runOnOperation() override {
+
+    using PatternFunction = llvm::function_ref<void(RewritePatternSet &)>;
+    // Order matters here.
+    llvm::SmallVector<std::pair<StringRef, PatternFunction>> patternMap = {
+        {"tan", populateExpandTanPattern},
+        {"sinh", populateExpandSinhPattern},
+        {"cosh", populateExpandCoshPattern},
+        {"asinh", populateExpandAsinhPattern},
+        {"acosh", populateExpandAcoshPattern},
+        {"atanh", populateExpandAtanhPattern},
+        {"powf", populateExpandPowFPattern},
+        {"fpowi", populateExpandFPowIPattern},
+        {"erf",
+         [&](RewritePatternSet &patterns) {
+           if (clNativeMathPrecision) {
+             patterns.add<math::ErfPolynomialApproximation>(&getContext());
+           } else {
+             populateExpandExp2FPattern(patterns);
+             populateMathPolynomialApproximationPatterns(patterns);
+             populateExpandRoundEvenPattern(patterns);
+           }
```

Comment:
This lambda should be a local variable to avoid taking references to temporaries.

Reply:
Thanks. I finally realized this is an issue. I updated the code.